arrows/mvg/applets/track_features.cxx

```diff
@@ -296,101 +244,225 @@ select_frames(std::vector<kwiver::vital::frame_id_t> const& valid_frames,
+        LOG_INFO(main_logger,
+          "Configuration file contained valid parameters and may be "
+          "used for running");
       }
       else
       {
         LOG_WARN(main_logger, "Configuration deemed not valid.");
       }
-      return EXIT_SUCCESS;
+      config = nullptr;
+      return WRITE;
     }
-    else if(!valid_config)
+    else if (!valid_config)
     {
       LOG_ERROR(main_logger, "Configuration not valid.");
-      return EXIT_FAILURE;
+      config = nullptr;
+      return FAIL;
     }
 
+    return SUCCESS;
+  }
+
+  // ------------------------------------------------------------------
+  kv::config_block_sptr default_config()
+  {
+    auto config = p->find_configuration("applets/track_features.conf");
+
+    // choose video or image list reader based on file extension
+    auto vr_config = config->subblock_view("video_reader");
+    if (ST::GetFilenameLastExtension(video_file) == ".txt")
+    {
+      vr_config->merge_config(
+        p->find_configuration("core_image_list_video_input.conf"));
+    }
+    else
+    {
+      vr_config->merge_config(
+        p->find_configuration("ffmpeg_video_input.conf"));
+    }
+
+    // choose video or image list reader for masks based on file extension
+    auto mr_config = config->subblock_view("mask_reader");
+    if (ST::GetFilenameLastExtension(mask_file) == ".txt")
+    {
+      mr_config->merge_config(
+        p->find_configuration("core_image_list_video_input.conf"));
+    }
+    else
+    {
+      mr_config->merge_config(
+        p->find_configuration("ffmpeg_video_input.conf"));
+    }
+
+    config->set_value("video_source", video_file,
+      "Path to an input file to be opened as a video. "
+      "This could be either a video file or a text file "
+      "containing new-line separated paths to sequential "
+      "image files.");
+
+    config->set_value("output_tracks_file", track_file,
+      "Path to a file to write output tracks to. If this "
+      "file exists, it will be overwritten.");
+
+    config->set_value("mask_source", mask_file,
+      "Optional path to an mask input file to be opened "
+      "as a video. "
+      "This could be either a video file or a text file "
+      "containing new-line separated paths to sequential "
+      "image files. "
+      "This list should be "
+      "parallel in association to frames provided by the "
+      "``video_source`` video. Mask images must be the same size "
+      "as the image they are associated with.\n"
+      "\n"
+      "Leave this blank if no image masking is desired.");
+
+    config->set_value("output_homography_file", homography_file,
+      "Optional path to a file to write source-to-reference "
+      "homographies for each frame. Leave blank to disable this "
+      "output. The output_homography_generator algorithm type "
+      "only needs to be set if this is set.");
+
+    config->set_value("invert_masks", false,
```

Comment:
@mleotta This is already in track-features.conf. Why do you set it again in default_config()

Reply:
It's not strictly necessary for parameters to be in both places.  The reason to have them in both places is primarily for documentation.  I like having all parameters defined here in the code so that a developer can see what all the parameters are in one location.  It also means that if the installed config files can't be found for whatever reason the executable can still output what all the expected parameters are.  I like having the parameters also defined in the installed config files because this is what is easiest for end users to look at, copy, and modify after KWIVER is installed.  We could leave parameters like this out of the config file and the defaults would be the same, but then end users would be less likely to notice that this parameter even exists.  It's common practice to copy these installed config files into your working directory and modify them.  Of course the downside of having these specified in two places is keeping them up to date and in sync.  That's a legitimate problem that I don't have a good answer for.

Reply:
Sounds good. Thanks Matt.